cni/network/network.go

```diff
@@ -779,47 +809,80 @@ func (plugin *NetPlugin) createEndpointInternal(opt *createEndpointInternalOpt)
 	}
 
 	if opt.nwCfg.MultiTenancy {
-		plugin.multitenancyClient.SetupRoutingForMultitenancy(opt.nwCfg, opt.cnsNetworkConfig, opt.azIpamResult, &epInfo, &defaultInterfaceInfo)
+		// previously only infra nic was passed into this function but now all nics are passed in (possibly breaks swift v2)
+		plugin.multitenancyClient.SetupRoutingForMultitenancy(opt.nwCfg, opt.cnsNetworkConfig, opt.azIpamResult, &epInfo, opt.ifInfo)
 	}
 
 	setEndpointOptions(opt.cnsNetworkConfig, &epInfo, vethName)
 
-	cnsclient, err := cnscli.New(opt.nwCfg.CNSUrl, defaultRequestTimeout)
+	// populate endpoint info with network info fields
+	epInfo.MasterIfName = nwInfo.MasterIfName
+	epInfo.AdapterName = nwInfo.AdapterName
+	epInfo.NetworkID = nwInfo.NetworkID
+	epInfo.Mode = nwInfo.Mode
+	epInfo.Subnets = nwInfo.Subnets
+	epInfo.PodSubnet = nwInfo.PodSubnet
+	epInfo.BridgeName = nwInfo.BridgeName
+	epInfo.NetNs = nwInfo.NetNs
+	epInfo.Options = nwInfo.Options
+	epInfo.DisableHairpinOnHostInterface = nwInfo.DisableHairpinOnHostInterface
+	epInfo.IPAMType = nwInfo.IPAMType
+	epInfo.IsIPv6Enabled = nwInfo.IsIPv6Enabled
+	epInfo.NetworkDNS = nwInfo.NetworkDNS
```

Comment:
put everything in an epinfo (do not initialize twice)-- init once only-- rename addSubnetToNetworkInfo to addSubnetToEndpointInfo (etc. anything to NetworkInfo to EndpointInfo)